react/components/Slider.js

```diff
@@ -1,7 +1,492 @@
+    const howManySliderToSlide = Math.ceil(movementDistance / (this.selectorWidth / this.perPage))
+
+    const slideToNegativeClone = movement > 0 && currentSlide - howManySliderToSlide < 0
+    const slideToPositiveClone = movement < 0 && currentSlide + howManySliderToSlide > this.innerElements.length - this.perPage
+
+    let newCurrentSlide = currentSlide
+    if (movement > 0 && movementDistance > threshold && this.innerElements.length > this.perPage) {
+      newCurrentSlide = this.prev(howManySliderToSlide)
+    } else if (movement < 0 && movementDistance > threshold && this.innerElements.length > this.perPage) {
+      newCurrentSlide = this.next(howManySliderToSlide)
+    }
+    this.slideToCurrent(slideToNegativeClone || slideToPositiveClone, newCurrentSlide)
+  }
+
+  _clearDrag = () => {
+    this.drag = {
+      startX: 0,
+      endX: 0,
+      startY: 0,
+      letItGo: null
+    }
+  }
+
+  onTouchStart = e => {
+    e.stopPropagation()
+    this.pointerDown = true
+    this.drag.startX = e.touches[0].pageX
+    this.drag.startY = e.touches[0].pageY
+  }
+
+  onTouchEnd = e => {
+    e.stopPropagation()
+    this.pointerDown = false
+    this.enableTransition()
+    if (this.drag.endX) {
+      this.updateAfterDrag()
+    }
+    this._clearDrag()
+  }
+  
+  onTouchMove = e => {
+    e.stopPropagation()
+
+    if (this.drag.letItGo === null) {
+      this.drag.letItGo = Math.abs(this.drag.startY - e.touches[0].pageY) < Math.abs(this.drag.startX - e.touches[0].pageX)
+    }
+
+    if (this.pointerDown && this.drag.letItGo) {
+      const { easing, loop, currentSlide } = this.props
+
+      e.preventDefault()
+      this.drag.endX = e.touches[0].pageX
+
+      const computedCurrentSlide = loop ? currentSlide + this.perPage : currentSlide
+      const currentOffset = computedCurrentSlide * (this.selectorWidth / this.perPage)
+      const dragOffset = this.drag.endX - this.drag.startX
+      const offset = currentOffset - dragOffset
+
+      setStyle(this._sliderFrame.current, {
+        WebkitTransition: `all 0ms ${easing}`,
+        transition: `all 0ms ${easing}`,
+        transform: `translate3d(${offset * -1}px, 0, 0)`,
+        WebkitTransform: `translate3d(${offset * -1}px, 0, 0)`
+      })
+    }
+  }
+
+  onMouseDown = e => {
+    e.preventDefault()
+    e.stopPropagation()
+    this.pointerDown = true
+    this.drag.startX = e.pageX
+
+    setStyle(this._sliderFrame.current, {
+      cursor: this.props.cursorOnMouseDown
+    })
+  }
+
+  onMouseUp = e => {
+    const { draggable, cursor } = this.props
+
+    e.stopPropagation()
+    this.pointerDown = false
+    this.enableTransition(draggable ? { cursor } : {})
+
+    if (this.drag.endX) {
+      this.updateAfterDrag()
+    }
+
+    this._clearDrag()
+  }
+
+  onMouseMove = e => {
+    const { easing, loop, currentSlide, draggable, cursorOnMouseDown } = this.props
+
+    e.preventDefault()
+    if (this.pointerDown && draggable) {
+      // TODO prevent link clicks
```

Comment:
can we "persist" the event of the click listener, and call `preventDefault` here? I don't know if it will work, but could be worth the try.

Reply:
I tried this and didn't work. That is something really weird happening here, I didn't know that things happens this way, but apparently `onClick` and `onClickCapture` events are fired together with `onMouseUp`, but the state of global variables used inside of them is the same as it was when `onMouseDown` was executed (I don't know if this explanation made any sense, so I did an example).

Before saying what is the example I have to explain something. To control the drag of the slider I use an object called `this.drag` that has the x and y positions of the mouse, and I update this object in the `onMouseMove` event, so it would make sense to check if `this.drag.endX !== 0` and if so, call `e.preventDefault(); e.stopPropagation();` in `onClickCapture`, but somehow this doesn't work.

Open this [workspace](https://stoppropagation--storecomponents.myvtex.com/), drag the slider and look at the console, you gonna see that `onMouseMove` event is fired before `onClickCapture`, but `onClickCapture` has the old state of `this.drag` object.

Here is the code used in this workspace (you probably won't need to see the code of shelf, but I gonna put here anyway):
1. [Slider](https://github.com/vtex-apps/slider/commit/27a20a4bb1401c5767e8190792060f90bdc0d8cb)
2. [Shelf](https://github.com/vtex-apps/shelf/blob/feature/new-slider/react/ShelfContent.js)
3. `product-summary` is linked, but I just apply a tachyons class to the root of the component

That's why I can't get it done. The condition to test it isn't working, so I'm accepting sugestions to fix this, because I have none right now :disappointed:

Reply:
@matheusps found something really interesting that is probably gonna work here, but I talked with @brunojdo and we think that is better to let this to after v1